src/common/vector_operations/vector_copy.cpp

```diff
@@ -34,11 +34,10 @@ static const ValidityMask &CopyValidityMask(const Vector &v) {
 	}
 }
 
-void VectorOperations::Copy(const Vector &source_p, Vector &target, const SelectionVector &sel_p, idx_t source_count,
-                            idx_t source_offset, idx_t target_offset) {
+void VectorOperations::CopyPartial(const Vector &source_p, Vector &target, const SelectionVector &sel_p,
+                                   idx_t source_count, idx_t source_offset, idx_t copy_count, idx_t target_offset) {
```

Comment:
Had to copy an X amount of tuples from `source`, not `source_count - source_offset`